sdk/python/build.sh

```diff
@@ -24,36 +24,120 @@
 #   apt-get update -y
 #   apt-get install --no-install-recommends -y -q default-jdk
 #   wget http://central.maven.org/maven2/io/swagger/swagger-codegen-cli/2.4.1/swagger-codegen-cli-2.4.1.jar -O /tmp/swagger-codegen-cli.jar
+#   # where /tmp/swagger-codegen-cli.jar in the wget line above is the same as the variable SWAGGER_CODEGEN_CLI_FILE defined below
+
+
+set -e
+set -o nounset
+set -o pipefail
+
+usage()
+{
+    echo "Usage: build.sh
+     ( [OUTPUT_PKG_FILE] | [-i, --install ] )
+     [-h help]
+
+    The -o flag and -i flag should not both be set at the same time.  No flags is equivalent to -o ./kfp.tar.gz
+
+    Examples:
+    build.sh                               # build and output a package, kfp.tar.gz, in this directory
+    build.sh ~/my_kfp_pgk.tar.gz           # build and output a package, my_kfp_pkg.tar.gz in your home dir
+    build.sh --install                     # install an editable copy of this kfp package in this directory
+    "
+}
+
+SWAGGER_CODEGEN_CLI_FILE=/tmp/swagger-codegen-cli.jar
+if [[ ! -f ${SWAGGER_CODEGEN_CLI_FILE} ]]; then
+    echo "You don't have swagger codegen installed.  See the comments on this file for install instructions."
+fi
+
+
+DO_INSTALL=false
+while (($#)); do
+    arg=$1; shift
+    case "$arg" in
+        -i | --install )
+                                DO_INSTALL=true
+                                ;;
+        -h | --help )           usage
+                                exit
+                                ;;
+        * )
+                                OUTPUT_FILE=$arg
+                                ;;
+    esac
+done
+
+if [ ! -z ${OUTPUT_FILE+x} ] && [ "$DO_INSTALL" = true ] ; then
+  usage;
+  exit 1;
+fi
+
+TMP_DIR=$(mktemp -d)
+# the directory containing this script
+THIS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
 
 get_abs_filename() {
   # $1 : relative filename
   echo "$(cd "$(dirname "$1")" && pwd)/$(basename "$1")"
 }
 
-target_archive_file=${1:-kfp.tar.gz}
-target_archive_file=$(get_abs_filename "$target_archive_file")
+OUTPUT_FILE=${OUTPUT_FILE:-kfp.tar.gz}
+OUTPUT_FILE=$(get_abs_filename "$OUTPUT_FILE")
+
+if $DO_INSTALL ; then
```

Comment:
Why do you run this code twice? There is another copy on https://github.com/kubeflow/pipelines/pull/1173/files#diff-feae8ebc1be1c937255731f029fffc53R113

Reply:
excellent catch